### PR TITLE
Use Spark integration on SDK

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Add Spark integration on SDK.
## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/constants/enums.py [hunk 1]
```diff
@@ -180,9 +181,15 @@ class ExecutionMode(str, Enum, metaclass=MetaEnum):
...
+class RuntimeType(str, Enum, metaclass=MetaEnum):
     AQUEDUCT = "aqueduct"
     AIRFLOW = "airflow"
     K8S = "k8s"
     LAMBDA = "lambda"
     DATABRICKS = "databricks"
+    SPARK = "spark"
+
+
+class SparkRuntimeType(str, Enum, metaclass=MetaEnum):
```
Comment (likawind) [inline sdk/aqueduct/constants/enums.py:193]: is this the canonical way to decide if an enum is within a subset? On top of my head I would do `SPARK_RUNTIM_TYPES = [RuntimeType.SPARK, RuntimeType.DATABRICKS]` or `def requireSpark(t RuntimeType)` instead

**Review by likawind — APPROVED** [review]
LGTM besides some nits

File: sdk/aqueduct/constants/enums.py (hunk 1 above)
Comment (kenxu95) [reply sdk/aqueduct/constants/enums.py:193]: We should add a unit test making sure that SparkRuntimeType is always a strict subset of RuntimeType. And another one making sure that RuntimeType is a subset of ServiceType.

File: sdk/aqueduct/models/config.py [hunk 2]
```diff
@@ -52,7 +58,13 @@ class Config:
 # TODO(...): this is deprecated.
 class FlowConfig(BaseModel):
     engine: Optional[
-        Union[AirflowIntegration, K8sIntegration, LambdaIntegration, DatabricksIntegration]
+        Union[
```
Comment (kenxu95) [inline sdk/aqueduct/models/config.py:61]: Can you add ENG-2511 into the deprecated TODO right above?

File: sdk/aqueduct/constants/enums.py (hunk 1 above)
Comment (kenxu95) [reply sdk/aqueduct/constants/enums.py:193]: @likawind The MetaEnum class allows us to use the `in` syntax - from what I remember reading it seemed like ok practice. I don't know if its canonical but it seems pythonic :P We should definitely add a comment here noting that it's a subset of RuntimeType though.

**Review by kenxu95 — COMMENTED** [review]
Can you add an entry for Spark in `connect_config` too? Please check if it's possible to connect via `client.connect_integration()` :)

**Pull request merged**